### PR TITLE
Add in a max_size to our redis backend

### PR DESCRIPTION
**Pull request by mattrobenolt**
@getsentry/infrastructure

---

This change is [Reviewable](https://reviewable.io/reviews/getsentry/sentry/3749)

### CODE REVIEW

File: src/sentry/cache/redis.py [hunk 1]
```diff
@@ -26,6 +27,8 @@ def __init__(self, **options):
     def set(self, key, value, timeout, version=None):
         key = self.make_key(key, version=version)
         v = json.dumps(value)
+        if len(v) > self.max_size:
+            raise ValueError('Cache key too large: %r %r' % (key, len(v)))
```
Comment (mattrobenolt) [inline src/sentry/cache/redis.py]: I'm not entirely sure of the ramification of raising a exception here vs dropping it silently, then logging.

I'm a bit afraid about raising, then the task getting retried.

**Comment by mitsuhiko** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/getsentry/sentry/3749#-KN3YqzcZs12O9OYzWP2:-KN3YqzcZs12O9OYzWP3:-1903478223) (no related file):*
I am okay with that. I would prefer a different exception class just so that we don't accidentally cathc it down somewhere. There are a few places where we catch down ValueError and TypeError over large blocks.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/getsentry/sentry/3749)*

**Comment by mitsuhiko** [comment]
Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/getsentry/sentry/3749)*

**Comment by mitsuhiko** [comment]
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/getsentry/sentry/3749)*

**Pull request merged**